src/docker-runner/utils.ts

```diff
@@ -0,0 +1,68 @@
+import execa from 'execa';
+import shell from 'shelljs';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+// Import { log } from '../log';
+
+export const isDockerInstalled = () => {
+  if (!shell.which('docker')) {
+    throw new Error('Docker is not installed');
+  }
+};
+
+const isLostPixelImageDownloaded = async ({ version }: { version: string }) => {
+  isDockerInstalled();
+  const { exitCode, stdout, stderr } = await execa('docker', [
+    'images',
+    '-q',
+    `lostpixel/lost-pixel:${version}`,
+  ]);
+
+  if (exitCode !== 0) {
+    throw new Error(`Not successful docker operation, ${stderr}`);
+  }
+
+  return stdout.trim().length > 0;
+};
+
+export const downloadImageIfNotExistent = async ({
+  version,
+}: {
+  version: string;
+}) => {
+  const doesImageExist = await isLostPixelImageDownloaded({
+    version,
+  });
+  if (!doesImageExist) {
+    await execa('docker', ['pull', `lostpixel/lost-pixel:v${version}`]);
+  }
+};
+
+type ParsedYargs = {
+  configDir: 'string';
+};
+
+export const executeDockerRun = async ({ version }: { version: string }) => {
+  await downloadImageIfNotExistent({ version });
+
+  // @ts-expect-error TBD
+  const argv = yargs(hideBin(process.argv)).parse() as ParsedYargs;
+
+  const args = [
+    'run',
+    '--rm',
+    '-v',
+    '$PWD:/workspace',
+    `--platform linux/amd64`,
```

Comment:
I am quite worried about this, not sure if it will not fail for non mac m1 users 🥲

Reply:
This needs to be evaluated based on the host OS.
Also, the images needs to be prepared for multi-platform use https://docs.docker.com/build/building/multi-platform/

Reply:
turns out it was just a warning, so this platform `linux/amd64` could be removed